### ISSUE TITLE
cls loss  is too large

### ISSUE THREAD

**Issue by SGQ-py** [zylo117/Yet-Another-EfficientDet-Pytorch#584]
Step: 499. Epoch: 0/10. Iteration: 500/1850. Cls loss: 117.17538. Reg loss: 2.39checkpoint...
Step: 999. Epoch: 0/10. Iteration: 1000/1850. Cls loss: 102.89307. Reg loss: 0.9checkpoint...
The Cls loss is too large,why? I checked the file of GT,it is right.Can you give me a answer?

**Comment by zylo117** [comment]
What is your learning rate?

**Comment by SGQ-py** [comment]
I never changed it, it's still 0.0001.


Thank you for your letter!


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/17/2021 22:32，zylo117<notifications@github.com> wrote：

What is your learning rate?

**Comment by zylo117** [comment]
try increasing it by 10 times and keep training for hours

**Comment by SGQ-py** [comment]
First of all, thank you very much for your prompt reply，I'll do it right away.


By the way, Could you please explain how to check if the GT input is correct during the training? I want to see how my approach differs from yours.


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/19/2021 23:25，zylo117<notifications@github.com> wrote：

try increasing it by 10 times and keep training for hours

**Comment by SGQ-py** [comment]
First of all, thank you very much for your prompt reply，I'll do it right away.


By the way, Could you please explain how to check whether the GT input is correct during the training? I want to see how my approach differs from yours.


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/19/2021 23:25，zylo117<notifications@github.com> wrote：

try increasing it by 10 times and keep training for hours

**Comment by zylo117** [comment]
by setting an argument `--debug True`, but there will be lots of false boxes if it's in the early stage of the training process.

**Comment by SGQ-py** [comment]
Wow! I changed the learning rate, it is working.The Loss can be lower.
Thank you !  Sincerely


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/19/2021 23:42，zylo117<notifications@github.com> wrote：

by setting an argument --debug True, but there will be lots of false boxes if it's in the early stage of the training process.

**Comment by SGQ-py** [comment]
Hello, although my loss during training is very small, when I use the trained model to evaluate, the AP50 is only 0.143. Can you help me analyze the possible reasons?


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/20/2021 23:17，wxxd<17806259143@163.com> wrote：
Wow! I changed the learning rate, it is working.The Loss can be lower.
Thank you !  Sincerely


| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/19/2021 23:42，zylo117<notifications@github.com> wrote：

by setting an argument --debug True, but there will be lots of false boxes if it's in the early stage of the training process.

**Comment by zylo117** [comment]
If you dont mind, you can share your dataset and I might train it for you. Related infomation is written on the readme.

**Comment by SGQ-py** [comment]
My pleasure!
     I put the packaged data set file in the BaiDuYun. The annotation is an xml file, because I am not sure if the json format I am converting to coco is correct.This will cause you more trouble, but I really can't do anything about the training results. Thank you again for your enthusiastic answers!
     It has more than 14,800 training sets and 2252 test sets. I wanted to send it to your github before, but found that it did not meet the requirements you set, and was afraid of causing unnecessary troubles, so I didn't upload it.




百度云链接：https://pan.baidu.com/s/1mKFjC-_rUInt8NvTFZ3zmQ 
提取码：f49u 




| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/21/2021 22:31，zylo117<notifications@github.com> wrote：

If you dont mind, you can share your dataset and I might train it for you. Related infomation is written on the readme.

**Comment by zylo117** [comment]
ok, but it may take a while

**Comment by SGQ-py** [comment]
Yes, I understand it completely. Then, I hope you will tell me the specific parameter settings what you set, thank you very much again!
| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/28/2021 18:31，wxxd<17806259143@163.com> wrote：
My pleasure!
     I put the packaged data set file in the BaiDuYun. The annotation is an xml file, because I am not sure if the json format I am converting to coco is correct.This will cause you more trouble, but I really can't do anything about the training results. Thank you again for your enthusiastic answers!
     It has more than 14,800 training sets and 2252 test sets. I wanted to send it to your github before, but found that it did not meet the requirements you set, and was afraid of causing unnecessary troubles, so I didn't upload it.




百度云链接：https://pan.baidu.com/s/1mKFjC-_rUInt8NvTFZ3zmQ 
提取码：f49u 




| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/21/2021 22:31，zylo117<notifications@github.com> wrote：

If you dont mind, you can share your dataset and I might train it for you. Related infomation is written on the readme.

**Comment by zylo117** [comment]
Just converted your dataset, the images are so blurry. The targets are barely visible. Have they been resized from a lower resolution or been blurred for some reason? I was thinking they are upsampled from 256x256 to 512x512 which means 75% pixels are redundant.
Anyway, I don't have high confidence that this can be trained well.
But for the large objects, I make it to mAP 0.149. As for the smaller blurrer objects, emmmmmmmm. I'd have given up on them.

I removed all the no-object images and trained 20k steps with lr 1e-3 and batchsize 8.
And this is the weights. Cls loss is about 1.7. It's not converged yet and can be further improved.

[efficientdet-d0_15_19564.zip](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/files/5899788/efficientdet-d0_15_19564.zip)

**Comment by SGQ-py** [comment]
First of all, I am very pleased with your letter. As you said, image blur is a problem of the data set itself, that is the image quality of the data set itself is poor, it has not been resized. To be honest, with the experiment doing, I have gradually lost confidence in the higher expectations of this data set. In related papers, I see that their indicators only use miss rate as the main evaluation method. There is not much mention of mAP value, although it is one of the most important indicators in the field of target detection.
    The data set comes from the 2015 Korea KAIST University Laboratory. 
    Anyway, this is maybe a trial and error direction. I am now trying to try other possible methods to see if there is a new breakthrough,such as transformer.
    I am no longer alone in communicating with you, thank you for your answers. Best wishes!
| |
wxxd
|
|
17806259143@163.com
|
签名由网易邮箱大师定制
On 1/31/2021 21:26，zylo117<notifications@github.com> wrote：

Just converted your dataset, the images are so blurry. The targets are barely visible. Have they been resized from a lower resolution or been blurred for some reason.
Anyway, I don't have high confidence that this can be trained well.
But for the large objects, I make it to mAP 0.149. As for the smaller blurrer objects, emmmmmmmm. I'd have given up on them.

I removed all the no-object images and trained 20k steps with lr 1e-3 and batchsize 8.
And this is the weights

efficientdet-d0_15_19564.zip
